logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/call/EndCallUseCaseTest.kt

```diff
@@ -55,5 +64,65 @@ class EndCallUseCaseTest {
             .function(callRepository::updateIsCameraOnById)
             .with(eq(conversationId.toString()), eq(false))
             .wasInvoked(once)
+
+        verify(callRepository)
+            .function(callRepository::persistMissedCall)
+            .with(eq(conversationId.toString()))
+            .wasNotInvoked()
+    }
+
+    @Test
+    fun givenNonEstablishedCall_whenEndCallIsInvoked_thenSaveMissedCall() = runTest {
+        given(callRepository)
+            .suspendFunction(callRepository::persistMissedCall)
+            .whenInvokedWith(eq(conversationId))
+            .thenDoNothing()
+
+        given(callRepository)
+            .suspendFunction(callRepository::establishedCallsFlow)
+            .whenInvoked().then { flowOf(listOf(call.copy(status = CallStatus.INCOMING))) }
+
+        given(callManager)
+            .suspendFunction(callManager::endCall)
+            .whenInvokedWith(eq(conversationId))
+            .thenDoNothing()
+
+        given(callRepository)
+            .function(callRepository::updateIsCameraOnById)
+            .whenInvokedWith(eq(conversationId.toString()), eq(false))
+            .thenDoNothing()
+
+        endCall.invoke(conversationId)
+
+        verify(callManager)
+            .suspendFunction(callManager::endCall)
+            .with(eq(conversationId))
+            .wasInvoked(once)
+
+        verify(callRepository)
+            .function(callRepository::updateIsCameraOnById)
+            .with(eq(conversationId.toString()), eq(false))
+            .wasInvoked(once)
+
+        verify(callRepository)
+            .suspendFunction(callRepository::persistMissedCall)
+            .with(any())
+            .wasNotInvoked()
```

Comment:
Shouldn't this be invoked per name of the test? 🤔

Reply:
the name of the test is: `givenNonEstablishedCall_whenEndCallIsInvoked_thenSaveMissedCall`, so shouldn't `callRepository::persistMissedCall` be invoked at least once here?

Reply:
Oh! yes good catch